code/modules/client/preference/character.dm

```diff
@@ -2029,6 +2029,10 @@
 			if(job.hidden_from_job_prefs)
 				continue
 
+			if(job.mentor_only)
+				if(!check_rights(R_MENTOR || R_ADMIN, FALSE, user))
```

Comment:
```suggestion
				if(!check_rights(R_MENTOR | R_ADMIN, FALSE, user))
```
Pretty sure you meant this
Don't forget to test your code with only mentor permissions on!